### DISCUSSION TITLE
Future of doctest in pytest and its impact on pytest-doctestplus

### DISCUSSION THREAD

**Discussion by pllim** [pytest-dev/pytest#8088] [General] [answered]
Hello! I have some questions regarding the following excerpt from https://docs.pytest.org/en/stable/doctest.html:

<hr/>

The builtin pytest doctest supports only doctest blocks, but if you are looking for more advanced checking over all your documentation, including doctests, `.. codeblock:: python` Sphinx directive support, and any other examples your documentation may include, you may wish to consider [Sybil](https://sybil.readthedocs.io/en/latest/index.html). It provides pytest integration out of the box.

<hr/>

The Astropy Project has a [pytest-doctestplus](https://github.com/astropy/pytest-doctestplus) plugin that was built on your doctest. Based on comments at https://github.com/astropy/pytest-doctestplus/issues/100#issuecomment-600946835 and those that followed, and now the statement above:

1. What is the future of doctest in pytest proper? Do you have a roadmap (either of future support or deprecation)?
2. Given that you endorsed Sybil above, do you work with Sybil developers? If we migrate our plugin to use Sybil, what is the risk of having to migrate again if it gets abandoned?

Hope you could advise. Thank you very much!

cc @cadair

**Comment by nicoddemus** [comment, marked as answer]
Hi @pllim,

Thanks for the thoughtful post.

> What is the future of doctest in pytest proper? Do you have a roadmap (either of future support or deprecation)?

We don't have any thoughts of deprecating it, and plan to keep maintaining it. At the same time, we don't have any plans for large improvements in the future.

> Given that you endorsed Sybil above, do you work with Sybil developers? If we migrate our plugin to use Sybil, what is the risk of having to migrate again if it gets abandoned?

We don't work with Sybil developers, we just mentioned it our own docs because seems well supported and a good alternative. I recall it was added based on a suggestion/PR from someone else in the community. 

The pytest team doesn't make any assurances about the future of Sybil.

Having said that, I think we could change the documentation there a bit: I think it would be more appropriate to have a separate section about external doctest support, and mention both Sybil and pytest-doctestplus as alternatives, with the disclaimer that there are no guarantees from the pytest team regarding those projects.

Let me know if there's anything else you would like to clarify!

**Reply by pllim** [reply]
Thank you!
